src/sync/deployment.test.ts

```diff
@@ -30,6 +37,221 @@ describe("sync/deployments", () => {
+
+				nockDeploymentCommitGetApi([deployments[3], deployments[2]], REPEAT_ONCE);
+
+				const result = await getDeploymentTask(logger, gitHubClient, jiraHost, repositoryData, DEPLOYMENT_CURSOR_EMPTY, PAGE_SIZE__TWO_ITEMS, msgPayload());
+
+				await expectDeploymentEntryInDB([deployments[3], deployments[2]]);
+				expectEdgesAndPayloadMatchToDeploymentCommits(result, [deployments[3], deployments[2]]);
+
+			});
+		});
+
+		describe("when ff is off", () => {
+
+			beforeEach(() => {
+				when(booleanFlag).calledWith(BooleanFlags.USE_DYNAMODB_FOR_DEPLOYMENT_BACKFILL, jiraHost).mockResolvedValue(false);
+			});
+
+			describe("for empty demployment cursor", () => {
+				// eslint-disable-next-line jest/expect-expect
+				it("should fetch deployments from begining", async () => {
+
+					const deployments = createDeploymentEntities(4);
+
+					nockFetchingDeploymentgPagesGraphQL(getDeploymentsQuery, DEPLOYMENT_CURSOR_EMPTY, [deployments[3], deployments[2]]);
+
+					nockDeploymentCommitGetApi([deployments[3], deployments[2]], REPEAT_ONCE);
+
+					//because ff is off, it won't find cache in history, so need to call the rest listing api lots of times
+					nockDeploymentListingApi(deployments, REPEAT_LOTS_OF_TIME);
+					nockDeploymentStatusApi(deployments, REPEAT_LOTS_OF_TIME);
+
+					const result = await getDeploymentTask(logger, gitHubClient, jiraHost, repositoryData, DEPLOYMENT_CURSOR_EMPTY, PAGE_SIZE__TWO_ITEMS, msgPayload());
+
+					expectEdgesAndPayloadMatchToDeploymentCommits(result, [deployments[3], deployments[2]]);
+				});
+			});
+
+			describe("for existing deployment cursor", () => {
+				// eslint-disable-next-line jest/expect-expect
+				it("should fetch deployments from existing cursor", async () => {
+
+					const deployments = createDeploymentEntities(4);
+
+					nockFetchingDeploymentgPagesGraphQL(getDeploymentsQuery, deployments[2].cursor, [deployments[1], deployments[0]]);
+
+					nockDeploymentCommitGetApi([deployments[1], deployments[0]], REPEAT_ONCE);
+
+					//because ff is off, it won't find cache in history, so need to call the rest listing api lots of times
+					nockDeploymentListingApi(deployments, REPEAT_LOTS_OF_TIME);
+					nockDeploymentStatusApi(deployments, REPEAT_LOTS_OF_TIME);
+
+					const result = await getDeploymentTask(logger, gitHubClient, jiraHost, repositoryData, deployments[2].cursor, PAGE_SIZE__TWO_ITEMS, msgPayload());
+
+					expectEdgesAndPayloadMatchToDeploymentCommits(result, [deployments[1], deployments[0]]);
+				});
+			});
+		});
+
+		//--helper funcs--
+		const msgPayload = () => ({
+			jiraHost,
+			installationId: DatabaseStateCreator.GITHUB_INSTALLATION_ID,
+			commitsFromDate: "2023-01-01T00:00:00Z"
+		});
+
+		//size up to 9 entities
+		const createDeploymentEntities = (size: number) => {
+			return "*".repeat(size).split("").map((_, idx)=> {
```

Comment:
We can improve performance here too: `return Array.from({ length: size }, (_, idx) => {`